### PR TITLE
Jbtm 3764 Re-license Narayana from LGPL to Apache-2.0

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.redhat.com/browse/JBTM-3764

CORE  AS_TESTS RTS XTS QA_JTA QA_JTS_OPENJDKORB  LRA 

This PR changes the license. We need to do it on all our repos but I'm raising the change for just narayana first. If the PR is approved then I will apply my scripts to our other repos.

I've split the task into multiple commits. This is in case there are PR comments that apply to a particular step so that I can update a step without affecting the other steps (I automated the majority of the diffs). But the primary reason for staging the change is to maintain the history of how we made the transition, this will help anyone wanting to audit the change.

### CODE REVIEW

**Comment by mmusgrov** [comment]
@marcosgopen @jmfinelli @moulalis This is a rather large PR so I didn't want to pick on someone to review it, but if anyone fancies self assigning that would be great!

File: CONTRIBUTING.md
Comment (marcosgopen) [inline CONTRIBUTING.md:36]: +1

**Comment by mmusgrov** [comment]
> Regarding your commit about adding "Copyright The Narayana Authors" I would prefer not to include it (even if I understand your reasons). I am afraid that other teams' contributors might feel a little bit 'excluded'. What do you think @tomjenkinson @jmfinelli ? But as I said I understand your reason (add authors declaration for java files - it's not needed but it deters people from adding their own) so I am ok if we want to have it.

There is no legal requirement to add copyright notices to contributions and I understand that they do not carry legal force behind them (attribution can be tracked by information contained in the git commit). So initially I preferred not to have them. But this leaves us open to someone copying the Narayana files into their own project and to then add their own Copyright statement. This circumstance would be difficult to track since the files would not contain any reference to Narayana.

But if a user did the same to a file that contains the text `Copyright The Narayana Authors` then it becomes clear where the file originally came from. If they still want to add their own Copyright under the `The Narayana Authors` then they can do that but we should discourage it since the act of committing new files effectively makes them one of the `Narayana Authors` so adding extra copyright notices becomes redundant and obfuscates. This reminds me that I forgot to add text asking people not to add `@author` tags to the files they create.

**Comment by mmusgrov** [comment]
> 
> ... BTW could I ask you to squash some of them together? "JBTM-3764 some XML files are missing SPDX ids" and "JBTM-3764 fix up one-line SPDX id declarations" could be squashed together IMO. Also "JBTM-3764 remove extra copyright on some vertx files" and "JBTM-3764 add authors declaration for java files - it's not needed but it deters people from adding their own" could be squashed I think.

I'll squash at the end because I used different scripts for updating the xml and java files.

I notice that you have withheld your approval - are you just blocked on the squashes and the Copyright notice or are you still working your way through the PR?

**Review by marcosgopen — APPROVED** [review]
After reading your comment and the document about license (https://source.redhat.com/departments/legal/redhatintellectualproperty/intellectual_property_and_ip_litigation_wiki/copyright_notices_in_source_code) I am ok with the copyright statement you added. If you planned to squash commits for this PR I think this PR is good and I approve it.

**Comment by mmusgrov** [comment]
@marcosgopen FYI I am starting on the quickstart repo this morning. That repo is designed for developers to copy into their own projects so I propose to not include a copyright in those files, ie I am going to remove all copyrights from the quickstart repo and just include a single license file and a license tag in the top level pom.

**Comment by marcosgopen** [comment]
Thank you @mmusgrov for addressing my request about squashing the commits. Your changes are ok, so your PR still has my approval.

**Pull request merged**
